### PR TITLE
Circleci

### PR DESCRIPTION
**Pull request by petercipriano-ms**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+#---------------------------------------------
+# weaviate Staging Jobs
+#---------------------------------------------
+  get_env_var_from_vault_sg_staging_eks:
+    <<: *docker-image
+    steps:
+      - checkout
+      - *prod_aws_creds
+      - moneysmart-orb/get_env_var_from_vault:
+          secret_path: staging/sg/weaviate
```
Comment (apoorvshrivastava-ms) [inline .circleci/config.yml]: vault file should be 
ms-labs-weaviate

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+          branch_name: master
+
+  get_env_var_from_vault_sg_prod_eks:
+    <<: *docker-image
+    steps:
+      - checkout
+      - *prod_aws_creds
+      - moneysmart-orb/get_env_var_from_vault:
+          secret_path: production/sg/weaviate
+          repo_name: weaviate-helm
```
Comment (apoorvshrivastava-ms) [inline .circleci/config.yml]: which repo is this ?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            
+            sed -i "s|QUERY_LIMIT|$QUERY_LIMIT|g" weaviate/values-override.yml
+            sed -i "s|AUTHENTICATION_API_KEY|$AUTHENTICATION_API_KEY|g" weaviate/values-override.yml
+            sed -i "s|'MODULES_API_KEY'|"$MODULES_API_KEY"|g" weaviate/values-override.yml
+            sed -i "s|SERVICE_TYPE|$SERVICE_TYPE|g" weaviate/values-override.yml
+      - helm/upgrade-helm-chart:
+          chart: weaviate
+          release-name: ms-labs-weaviate
+          namespace: labs-db
+          values-to-override: image.tag=1.23.0,replicas=3,storage.size=50Gi
```
Comment (apoorvshrivastava-ms) [inline .circleci/config.yml]: can these come from vault ?
atleast the replicas and storage size ?
and image tag from vaule.yml ?

File: .circleci/config.yml (hunk 2 above)
Comment (petercipriano-ms) [reply .circleci/config.yml]: Will rename to ms-labs-weaviate

File: .circleci/config.yml (hunk 3 above)
Comment (petercipriano-ms) [reply .circleci/config.yml]: Sure we can add the replicas and storage size to vault

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            
+            sed -i "s|QUERY_LIMIT|$QUERY_LIMIT|g" weaviate/values-override.yml
+            sed -i "s|AUTHENTICATION_API_KEY|$AUTHENTICATION_API_KEY|g" weaviate/values-override.yml
+            sed -i "s|'MODULES_API_KEY'|"$MODULES_API_KEY"|g" weaviate/values-override.yml
+            sed -i "s|SERVICE_TYPE|$SERVICE_TYPE|g" weaviate/values-override.yml
+      - helm/upgrade-helm-chart:
+          chart: weaviate
+          release-name: ms-labs-weaviate
+          namespace: labs-db
+          values-to-override: appLabels.region=sg,image.tag=1.23.0
```
Comment (apoorvshrivastava-ms) [inline .circleci/config.yml]: please move image tag to values.yml ?

**Review by apoorvshrivastava-ms — COMMENTED** [review]
Can we make sure proper annotations are added to ensure 1 pod is always placed on OD instance

File: weaviate/values-override.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    allowed_keys:
+      - {{ .Values.envs.AUTHENTICATION_API_KEY }}
+    users:
+      - admin
+
+query_defaults:
+  limit: {{ .Values.envs.QUERY_LIMIT }}
+
+service:
+  name: weaviate
```
Comment (apoorvshrivastava-ms) [inline weaviate/values-override.yml]: this should be named ms-labs-weaviate
and should be accordingly updated in other places to match lables

**Review by apoorvshrivastava-ms — APPROVED** [review]

**Pull request merged**